### ISSUE TITLE
Gibberish code (Math.floor(user.stats.exp) etc) on homepage / tasks page prevents viewing tasks

### ISSUE THREAD

**Issue by innus** [HabitRPG/habitica#5473]
Heya.  I have no idea if this is the right place to put this, please let me know if not.  That said, the HabitRPG homepage looks like this for me:

![screenshot 2015-06-23 at 8 54 40 pm](https://cloud.githubusercontent.com/assets/13026274/8321328/06c55ebc-19eb-11e5-89a1-a83e9fedde9e.png)

If this has already been brought up and fixed, I'm sorry.  Please let me know if you need more info.

Thanks!

_edit by admin:_
# What to do if you experience this bug:

This bug seems to be caused by extensions that block or otherwise affect Google Analytics features (e.g., adblockers, privacy tools, tracker blockers, spyware blockers, etc). We are trying to prevent this happening so that you can keep using those tools. In the meantime, please try any one of the following things (you don't need to do them all):
1. Disable all extensions. This should fix the problem immediately. You can then, if you wish, enable extensions one by one until you find the one(s) that caused the problem.
2. Use HabitRPG in incognito/privacy mode (this will only work if the relevant extension(s) have not been configured to run in incognito mode; if you have configured them to do that, turning off that setting will allow incognito mode to work).
3. Use HabitRPG on another browser that does not have the extensions installed.

The extensions that might be causing the issue could be:  
- Google Analytics Opt-out Add-on
- Avira
- Privacy Badger
- WebGL Inspector
- Adblock
- Adblock Plus
- Disable Anti-Adblock
- and maybe others

These workaround should be only temporary! Once we work out what interaction is causing the problem, we should be able to modify HabitRPG so that you can keep using all extensions, even on HabitRPG itself.

_NB: You do not need to read on below unless you are curious about the details of the investigation._

**Comment by Alys** [comment]
Sorry about this! Could you please try reloading a couple of times (maybe over a few minutes if the first reload doesn't help) and let us know if that fixes it?

**Comment by innus** [comment]
Hi Alys!  Thanks for your quick reply!  I have actually been checking for about an hour because I'm hoping to fix it before midnight so I don't lose health (blush).  I just reloaded 3 more times and it's still a mess.  Sorry!

**Comment by innus** [comment]
Oh!  I've also tried rebooting the browser and the computer, and disabling my adblocker and privacy manager.

**Comment by Alys** [comment]
If you can give me your User ID (a string that looks like d904bd62-da08-416b-a816-ba797c9ee265) then I can make you [Rest in the Inn](http://habitrpg.wikia.com/wiki/Rest_in_the_Inn) to protect you, if you'd like that. You can get it from https://habitrpg.com/#/options/settings/api . You can also rest in the inn using the mobile app if you have it installed (it's under Settings).

If you can't get to that page and don't have the app, and do want me to make you rest in the inn, tell me the name or email address you use to log in, or your Facebook email address if you use facebook registration. Feel free to email it to admin@habitrpg.com if you don't want to display it publicly here in GitHub.

**Comment by innus** [comment]
Thank you so much.  I used the mobile app to rest in the inn.

**Comment by Alys** [comment]
Excellent! 

In theory, there should be nothing on the server causing this, so it might be a problem with your account. Can you please get your User ID from the app? You can find it under Settings and there's a button to copy it to the clipboard.

**Comment by innus** [comment]
Sure thing. It's ccec16ed-5fd4-4162-b6c0-273dea4aa1dd

**Comment by Alys** [comment]
This problem has been reported in the past (https://github.com/HabitRPG/habitrpg/issues/4847) and clearing the browser's cache and local storage fixed it then. Can you please try both of those? If you don't know how to clear local storage, google for the name of your browser with the words `clear "local storage"`

**Comment by innus** [comment]
Certainly!  Trying now.

**Comment by innus** [comment]
Hi again.  I'm so sorry, I cleared everything and it still looks the same.

**Comment by Alys** [comment]
Don't be sorry! This isn't your fault. I'm sorry it's happening. :(  

I haven't yet seen anything wrong in your account that would cause this.

I want to check that local storage really is cleared because not all clearing methods do that correctly. Can you please log out of HabitRPG if you are able to, then click on Chrome's Settings button, then on "More tools", and then on "JavaScript Console", and then on the "Resources" tab in the grey bar at the top of the frame that appears, and then click the arrow next to "Local Storage" in the left-hand pane. If the HabitRPG url appears under "Local Storage", click on that URL. Does anything appear in the right-hand pane? If it does, highlight each item in turn and click the "x" at the bottom of the pane, until the pane is empty. Then reload the site and log back in.

**Comment by innus** [comment]
Unfortunately, while I was waiting I went through the #4847 thread and tried everything there and everything linked there, including opening the console and clearing from there and then rebooting everything again.  I'm on a Chromebook if that makes a difference, but I was able to get on fine this morning.

**Comment by innus** [comment]
Not sure if this will help, but here is the info the console gives me about the HabitRPG page when it's loaded:

TypeError: Cannot read property 'childNodes' of undefined
    at pa (app-34283b84.js:6)
    at r (app-34283b84.js:5)
    at pa (app-34283b84.js:6)
    at r (app-34283b84.js:5)
    at pa (app-34283b84.js:6)
    at r (app-34283b84.js:5)
    at h (app-34283b84.js:5)
    at app-34283b84.js:4
    at j.$eval (app-34283b84.js:7)
    at j.$apply (app-34283b84.js:7)(anonymous function) @ app-34283b84.js:6
2connect.facebook.net/en_US/sdk.js&version=v2.0:64 Received message of type object from https://platform.twitter.com, expected a string
ping:1 Given URL is not allowed by the Application configuration: One or more of the given URLs is not allowed by the App's settings.  It must match the Website URL or Canvas URL, or the domain must be a subdomain of one of the App's domains.

**Comment by hairlessbear** [comment]
@innus Have you tried loading HabitRPG with none of your extensions loaded? You may be able to do that by switching to incognito mode.

Also, have you tried it in a different browser?

Finally, try the following method for clearing your local data from HabitRPG (of course, if you've already done this, feel free to ignore the following :smile:):
1. Go to Settings
2. Click "Show advanced settings" (at the bottom of the page)
3. Under the "Privacy" heading, click on "Content settings"
4. Click "All cookies and site data"
5. Enter "habitrpg" into the search bar; this should bring up the "habitrpg.com" entry
6. Click the `X` on the right side of the entry. This should delete all local data associated with HabitRPG, which will mean you'll have to log in again.
7. Log in again and report your results! If we're lucky, this might fix the problem.

**Comment by innus** [comment]
Done.  It works just fine in incognito mode, but even after deleting all local data, cookies, and everything else three times in different ways and then rebooting each time, it still doesn't work in normal mode.  I'm thinking I might need to (gasp) powerwash my Chromebook.  Any other suggestions before I lose all settings?  At least there's no data to lose since everything's on Google Drive.

**Comment by hairlessbear** [comment]
If it works in incognito but not in normal mode, that means that one (or more) of your extensions is messing with the page. I'd wager a guess that it's one that's designed to block certain types of content. In particular, anything that blocks scripts (ScriptSafe, ScriptBlock etc.) is a particularly likely culprit. Before you wipe everything, I'd suggest going through your list of extensions, and disabling ones that seem like likely culprits, one at a time, and refresh the page each time. Eventually, the page should work, and then you'll know which extension is causing the issue.

**Comment by innus** [comment]
Will do, hairlessbear!  Thank you so much and also thank you, Alys, for your help tonight.  After I check extensions I'm racking out, but before I do I'll report back on which extension was causing the problem if I'm able to find it.  Thanks again!

**Comment by hairlessbear** [comment]
:+1: Glad to help. And thank you in advance for reporting which extension is causing the issue. That could very well help us troubleshoot issues that future users have!

**Comment by innus** [comment]
The problem extension is... drumroll... Google Analytics Opt-out Add-on (by Google).  There are no settings available that I can find, so I can't set an exception for HabitRPG.  Interestingly, I've had it installed for weeks and HabitRPG has been working fine, but this afternoon I enabled 2-factor authentication for Google, and I think that's what did it.  

After a bit of research and thought, I remembered when I had a blog, I _loved_ seeing my analytics, and although I usually opt out of and block everything possible, I would hate to deny anyone the joy of tons of visits, so instead of searching all night for a workaround, I just uninstalled it.  I hope this can help others who are having this problem!

**Comment by Alys** [comment]
@innus Thank you very much for the details!

**Comment by mimicoflorido** [comment]
I was having this same problem, I solved it thanks to innus's suggestion that Google Analytics was the problem. I don't remember changing nothing myself. Something about Google Analytics must have changed a few hours back and that is what is bugging habitrpg for some of us.

Thanks for the help!

**Comment by hairlessbear** [comment]
@mimicoflorido Glad to hear it!

@innus See? I told you the information would be useful :smile:

**Comment by innus** [comment]
@mimicoflorido I'm so happy this helped!  

@hairlessbear @Alys Everyone on this thread has been so amazingly nice, I'm definitely going to subscribe.  Thanks again!

**Comment by mimicoflorido** [comment]
@innus Truly thanks!

I just found out another problem, thou. I fixed it in Opera by enabling Google Analytics in Ghostery, but in Firefox I didn't have any tracker blocker, so I found out it was the Avira extension! I wonder if Avira is blocking Google Analytics? It's weird, why would Avira block it?

**Comment by typpo** [comment]
Having the same problem starting this evening.

Chrome 43.0.2357.65  Adblockers disabled both on the page and in chrome://extensions.  User ID `ae7bd2ab-c0ba-4a0c-b89a-7507008ee836`.

No console output other than the following:

```
TypeError: Cannot read property 'childNodes' of undefined
at pa (app-34283b84.js:6)
at r (app-34283b84.js:5)
at pa (app-34283b84.js:6)
at r (app-34283b84.js:5)
at pa (app-34283b84.js:6)
at r (app-34283b84.js:5)
at h (app-34283b84.js:5)
at app-34283b84.js:4
at j.$get.j.$eval (app-34283b84.js:7)
at j.$get.j.$apply (app-34283b84.js:7)
```

However, it does work in incognito with uBlock enabled!  Which is confusing.

**Comment by innus** [comment]
@mimicoflorido I'm not sure, I've never used Avira, but I just looked it up and it says it blocks trackers, so that might be the problem.

**Comment by innus** [comment]
@typpo Do you have the Google Analytics Opt-out Add-on installed, or any tracking blockers like Privacy Badger?

**Comment by mimicoflorido** [comment]
@innus Oh, it explains. Thank you.

@typpo maybe it's the way trackers work in incognito...? @innus said the site worked in incognito.

**Comment by Alys** [comment]
Extensions (including trackers) by default do not work in incognito, but it is possible to set the extensions so that they do. If you've done that for any extensions relevant to the problem, then incognito mode won't help until you untick that setting. Please reply if you have questions about this.

Also, please see the first post in this thread. I've updated it with some generic tips for working around this problem until we fix it properly.

**Comment by gisikw** [comment]
Hey folks - we've recently made some changes to make the analytics run more smoothly, but it looks like it's ended up causing other parts of the site to fail when it's blocked. I'm investigating now, and should be able to get a fix up shortly. Appreciate your patience and help in identifying the issue!

**Comment by typpo** [comment]
@Alys I explicitly allowed both Disconnect and uBlock (my adblocking extensions) in incognito, like [this](http://i.imgur.com/FEQ8liA.png).  The page loads with both blockers ([pic](http://i.imgur.com/JOF8jmn.png)).  

I figured the explanation would be that something else is blocking GA.  So after a meticulous binary search through my extensions, I discovered that [WebGL Inspector](https://chrome.google.com/webstore/detail/webgl-inspector/ogkcjmbhnfmlnielkjhedpcjomeaghda) is the culprit!  I can't think of a good reason why..but there you go.

**Comment by Thor333** [comment]
Hi guys, same problem for me here. Only top bar showing, but worked fine in private mode.
I tried to refresh and all. I'm on Firefox 38.0.5, had the plugins Adblock plus 2.6.9.1 and Disable Anti-Adblock 3.1.1.
I tried to disconnect plugins one by one, then activating all of them, and it solved the problem...
Everything back like it was before, but i have everything back to normal now.
No problems for me any more, just witnessing.
Keep on the good work!

**Comment by Alys** [comment]
Thanks @Thor333!

**Comment by Alys** [comment]
Thanks very much to @gisikw's dedicated work over the past three hours, we have a fix now, although it's not live yet. We're going to do some brief testing on a pre-production server and then push it to the real site (that might be in about an hour because the deployments can take some time). I'll post here again when it's fixed on the real site.

**Comment by Alys** [comment]
This should now be properly fixed on the website. You should be able to enable all extensions and the site will work in normal mode (i.e., incognito mode is not needed). You'll need to reload the site (hopefully only once).

I'm closing this now but please comment here if you're still having trouble and we'll reopen and continue fixing it.

Thank you very much @innus @hairlessbear @mimicoflorido @typpo @Thor333 for your troubleshooting help and information about your extensions! It was very useful in working out what was going wrong.
